addons/website/models/res_lang.py

```diff
@@ -14,20 +15,23 @@ def write(self, vals):
                 raise UserError(_("Cannot deactivate a language that is currently used on a website."))
         return super(Lang, self).write(vals)
 
-    @api.model
     @tools.ormcache_context(keys=("website_id",))
-    def get_frontend_langs(self):
+    def _get_frontend(self) -> LangDataDict:
+        """ Return the available languages for current request
+        :return: LangDataDict({code: LangData})
+        """
         if request and getattr(request, 'is_frontend', True):
-            langs = self.env['website'].get_current_website().language_ids.get_sorted()
+            langs_ids = set(self.env['website'].get_current_website().language_ids.ids)
+            langs = [dict(lang) for lang in super()._get_frontend().values() if lang.id in langs_ids]
```

Comment:
I understand this override seems more complicated than before.
but it do have advantages. it respects more to the object-oriented design by calling `super()`
1. . This method is overridable. Every override method is allowed to remove some active languages but never increase
2. I use the result of `super()` to respect the order of the languages defined in its `super`, so that I can remove the `get_sorted` api

Reply:
I don't agree, we don't want the `super()` to mess up with the langs and we don't want the website languages ids to be filtered by the super (or the super to be filtered by the website language ids).
Once you have installed website, you define the frontend langs on the website, and nothing should be altering it.

Eg, if later the portal/http_routing says that the frontend langs are a subset of active langs and not all active langs, and then you install website, the super langs could not contains all the website langs, and some of the website langs would not be shown on the website.

We really don't want to consider the `super()` here.